### PR TITLE
Fix #6511: genesis magin plot

### LINKED ISSUE #6511: plot of genesis maginfile
It would be useful to visualize the maginfile used by a genesis simulation. 
Something similar to this:

Figure 1 (18)

### PR DESCRIPTION
**Pull request by gurhar1133**

### CODE REVIEW

**Comment by gurhar1133** [comment]
@moellep ping

File: sirepo/package_data/static/js/genesis.js [hunk 1]
```diff
@@ -56,6 +59,32 @@ SIREPO.app.directive('appFooter', function() {
...
+        controller: function($scope) {
+            const checkForMaginfile = () => {
+                return appState.models.io.maginfile || false;
+            };
+
+            $scope.hasMaginfile = checkForMaginfile();
+            $scope.$on('io.changed', ()=> {
+                $scope.hasMaginfile = checkForMaginfile();
+            });
+        }
```
Comment (moellep) [inline sirepo/package_data/static/js/genesis.js:84]: You could replace all the controller code by adding: 
```javascript 
    $scope.hasMaginfile = () => appState.applicationState().io.maginfile;
```
And changing the line to `data-ng-if="hasMaginfile()"`

Also, add a `data-report-id="reportId"` to the parameter-plot, and `$scope.reportId = utilities.reportId();` so you get the popup menu when you click on the plot.

File: sirepo/template/genesis.py [hunk 2]
```diff
@@ -382,12 +414,22 @@ def _get_frame_counts(run_dir):
...
-def _is_text_file(path):
-    try:
-        pkio.read_text(path)
-        return True
-    except UnicodeDecodeError:
-        return False
+def _parse_maginfile(filepath):
+    if not pkio.is_pure_text(filepath):
+        raise AssertionError(f"{filepath.basename} for maginfile should be text file")
+    p = []
```
Comment (moellep) [inline sirepo/template/genesis.py:420]: Also add a check that the file is pure text in template.genesis validate_file().

File: sirepo/template/genesis.py [hunk 3]
```diff
@@ -382,12 +414,22 @@ def _get_frame_counts(run_dir):
...
+    u = 1
+    with pkio.open_text(filepath) as f:
+        for line in f:
+            row = line.split()
+            if row:
+                if row[0] == _MAGIN_PLOT_FIELD:
+                    p.append(row[1])
+                if row[0] == "?" and "UNITLENGTH" in row[1]:
+                    u = row[2]
+    if p:
```
Comment (moellep) [inline sirepo/template/genesis.py:430]: This fails if there is a space after UNITLENGTH, for example, if the data file has this header:
```
? UNITLENGTH = 0.032
```
vs 
```
? UNITLENGTH= 0.032
```

**Review by moellep — COMMENTED** [review]
Looks good - I added a few comments.

File: sirepo/package_data/static/js/genesis.js (hunk 1 above)
Comment (gurhar1133) [reply sirepo/package_data/static/js/genesis.js:84]: Screenshot 2024-02-09 at 4 00 48 PM
This popup?

File: sirepo/template/genesis.py (hunk 2 above)
Comment (gurhar1133) [reply sirepo/template/genesis.py:420]: I thought I did. I changed the checks to `if pkio.is_pure_text(path):`

File: sirepo/template/genesis.py (hunk 3 above)
Comment (gurhar1133) [reply sirepo/template/genesis.py:430]: Oh, yeah. Thanks for catching this

File: sirepo/template/genesis.py (hunk 2 above)
Comment (moellep) [reply sirepo/template/genesis.py:420]: In template.genesis validate_file() should return an error if a user tries to upload a binary maginfile. There are already checks there for partfile and distfile.

**Review by moellep — APPROVED** [review]

**Pull request merged**